tests/test_nock_lifecycle.js

```diff
@@ -106,6 +106,21 @@ describe('Nock lifecycle functions', () => {
         return true
       })
     })
+
+    it('should be safe to call in the middle of a request', done => {
```

Comment:
What is it that shouldn't happen, exactly?

Reply:
This was accidentally left in from other work that required coverage. 
I'll remove it.

Reply:
I spoke too soon, this is still needed for coverage. Let me try to get some data on why, that might help explain or transform the test to have it make more sense.